src/components/LocationSelector/LocationSelectorDialog/LocationSelectorDialog.js

```diff
@@ -66,9 +74,15 @@ export const LocationSelectorDialog = (props) => {
         onUpdateLocation(firstSelectedLocation.id, radius);
     }
 
+    function handleInputFormSubmit(e) {
+        onCloseModal();
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
     return (
         <>
-            <div {...elem('inputLine', props)}>
+            <form {...elem('inputLine', props)} onSubmit={handleInputFormSubmit}>
```

Comment:
I think it is not a good idea to wrap the component into form just for a closing popup. 
Finally, we will have in Reach not valid HTML https://stackoverflow.com/questions/555928/is-it-valid-to-have-a-html-form-inside-another-html-form